internal/plugins/golang/v2/config.go

```diff
@@ -14,16 +14,20 @@
 
 package v2
 
-import "sigs.k8s.io/kubebuilder/v2/pkg/model/config"
+import (
+	"sigs.k8s.io/kubebuilder/v3/pkg/config"
+	cfgv3 "sigs.k8s.io/kubebuilder/v3/pkg/config/v3"
+)
 
 // Config configures this plugin, and is saved in the project config file.
 type Config struct{}
 
-// hasPluginConfig returns true if cfg.Plugins contains an exact match for this plugin's key.
-func hasPluginConfig(cfg *config.Config) bool {
-	if !cfg.IsV3() || len(cfg.Plugins) == 0 {
+// hasPluginConfig returns true if cfg.Layout contains an exact match for this plugin's key.
+func hasPluginConfig(cfg config.Config) bool {
+	isV3 := cfg.GetVersion().Compare(cfgv3.Version) >= 0
+	if !isV3 {
 		return false
 	}
-	_, hasKey := cfg.Plugins[pluginConfigKey]
-	return hasKey
+	var info struct{}
+	return cfg.DecodePluginConfig(pluginConfigKey, &info) == nil
```

Comment:
This will always return `nil` if the plugin key isn't present. As of now I don't think there's a way to check if a config has a plugin key. One should be added, ex `cfg.HasPluginKey(key)`, or this method should return an error if key isn't found.

Until this issue is fixed, this PR should be blocked.

Reply:
Oh true, should we actually return an error if a key is requested and not found?

Reply:
Yeah I think returning a typed error is sufficient so it can be type-checked by the caller, ex
```go
type PluginKeyNotFoundError struct {
	Key string
}

func (e PluginKeyNotFoundError) Error() string {
	return fmt.Sprintf("config.plugins does not contain key %q", e.Key)
}
```

Reply:
Done, https://github.com/kubernetes-sigs/kubebuilder/pull/1985